src/uu/hashsum/src/hashsum.rs

```diff
@@ -546,6 +547,10 @@ where
                     )
                 )
                 .to_ascii_lowercase();
+                // FIXME: (How) should these be quoted?
```

Comment:
I took a peek at  [GNU's implementation](https://github.com/coreutils/coreutils/blob/669d45de0f43861c3e690c82f0e6abd9174e2388/src/md5sum.c#L784), and they only escape `\\` and `\n`, if the file name contains a newline here. In a different place they escape the file name if it contains a newline or a backslash. Also, a backslash is prepended to the file name
The manual says:
```plaintext
Without --zero, if file contains a backslash or newline, the line is started with a backslash, and each problematic character in the file name is escaped with a backslash, making the output unambiguous even in the presence of arbitrary file names
```

Reply:
Thank you. I don't think I'll implement it in this PR, but I'll update the note.

What's the exact policy on looking at GNU code? I know using it as a reference is not allowed (#1994) but I have heard of other projects having one person read and describe the behavior and having someone else implement it (sort of like we're doing now).

Reply:
cc @sylvestre

I'm not sure what the exact policy is, but I have looked at the GNU source code in the past when I couldn't understand its behavior (i.e. when it did someting different from what it claimed to be doing in the manual) and I have submitted 3 bug reports to GNU coreutils (+1 to glibc) because of that already. So, my understanding is that it's fine to read the code to understand its behavior but not to copy the implementation details.

Reply:
I don't think it is a problem to look at the GNU code to understand if they add a newline or not.
It would be a problem if the code structure is taken from GNU's which isn't, AFAIK, the case.
AFAIK, we are quite different.